test/app/presenter/ChessBoardPresenter.spec.ts

```diff
@@ -42,6 +42,33 @@ describe('ChessBoardPresenter', () => {
 
     expect(view.showChessBoard).toHaveBeenCalledWith(PIECES_START_POSITION);
   });
+
+  it('new test for view move method (later the test name will be changed)', () => {
+    const pawn: Pawn = new Pawn(Side.WHITE);
+    const viewEvents: ViewEventBus = new InMemoryViewEventBus();
+    const view: ChessBoardView = chessBoardViewMock(viewEvents);
+    const model: ChessModel = {
+      squaresWithPiece: PIECES_START_POSITION,
+      move: (squareFrom, squareTo) => {
+        return [{ eventType: 'PieceWasMoved', piece: pawn, from: squareFrom, to: squareTo }];
+      },
+      possibleMoves: jest.fn(),
+    };
+    const presenter: ChessBoardPresenter = new ChessBoardPresenter(view, model);
+    // const chessboardState = chessboardStateMock([
+    //   { column: 'A', row: 3 },
+    //   { column: 'A', row: 4 },
+    // ]);
+
+    // chessboardState.viewEvents.publish(new SquareWasClicked({ x: 1, y: 2 }));
+    // chessboardState.viewEvents.publish(new SquareWasClicked({ x: 1, y: 3 }));
+    model.move({ column: 'A', row: 2 }, { column: 'A', row: 3 });
```

Comment:
Prawie dobrze, tego na pewno tutaj nie powinno być :) Presenter traktujemy jako taką czarną skrzynkę. Przyjmuje z widoku jakieś zdarzenia (SquareWasClicked) i reaguje na nie odpowiednio. Czyli np. w tym przypadku musisz tymi squareWasClicked zasymulowac dokladnie to co Ty robisz na boardzie. 
0. Zakładasz co będzie zwrócone zmetod modelu, które są Ci potrzebne - mockujesz tak jak zrobiłeś np. move. PossibleMoves tez chyba trzeba?
1. Kliknac pionka. (`viewEvents.publish(new SquareWasClicked({ x: 1, y: 2 }))`)
2. Kliknac na miejsce gdzie chesz go przenieśc (także mówisz, że to zrobiłeś za pomocą eventu, w teście view jest sprawdzanie czy widok generuje te eventy).
3. Sprawdzasz czy na widoku zostaly wywolane wlasciwe metody, w stylu capturePiece() / movePiece().